### PR TITLE
Change minimum timestep size to 1e-12

### PR DESCRIPTION
**Pull request by totto82**
the unit is days for field and metric and hours for lab
the purpose is to shut down "dead" simulations

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

File: opm/simulators/timestepping/AdaptiveTimeSteppingEbos.hpp [hunk 1]
```diff
@@ -693,6 +682,21 @@ namespace Opm {
...
+                case UnitSystem::UnitType::UNIT_TYPE_FIELD:
+                case UnitSystem::UnitType::UNIT_TYPE_METRIC:
+                    return value * unit::day;
+                    break;
+                case UnitSystem::UnitType::UNIT_TYPE_LAB:
+                    return value * unit::hour;
+                    break;
+                default:
+                    throw std::runtime_error("Unsupported unit type when initializing the time stepper");
+            }
```
Comment (bska) [inline opm/simulators/timestepping/AdaptiveTimeSteppingEbos.hpp:697]: Is there a reason you did not write this as
```C++
return unitSystem.to_si(UnitSystem::time, value);
```

Comment (totto82) [reply opm/simulators/timestepping/AdaptiveTimeSteppingEbos.hpp:697]: I just copied @GitPaean code. Maybe he had a reason? If not I can update accordingly.

Comment (GitPaean) [reply opm/simulators/timestepping/AdaptiveTimeSteppingEbos.hpp:697]: There is no good reason. That function was not in my mind.  I think @bska's suggestion is good.

File: opm/simulators/timestepping/AdaptiveTimeSteppingEbos.hpp [hunk 2]
```diff
@@ -312,7 +312,7 @@ namespace Opm {
             EWOMS_REGISTER_PARAM(TypeTag, double, SolverMaxTimeStepInDays,
                                  "The maximum size of a time step in days");
             EWOMS_REGISTER_PARAM(TypeTag, double, SolverMinTimeStep,
-                                 "The minimum size of a time step in seconds. If a step cannot converge without getting cut below this step size the simulator will stop");
+                                 "The minimum size of a time step in days for field and hours for lab. If a step cannot converge without getting cut below this step size the simulator will stop");
```
Comment (GitPaean) [inline opm/simulators/timestepping/AdaptiveTimeSteppingEbos.hpp:315]: For completion, maybe metric should be mentioned here?

File: opm/simulators/timestepping/AdaptiveTimeSteppingEbos.hpp (hunk 1 above)
Comment (totto82) [reply opm/simulators/timestepping/AdaptiveTimeSteppingEbos.hpp]: It was not in my mind either. Thanks to @bska for pointing it out.

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
benchmark please

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test    | Configuration                                     | Relative |
| ------- | ------------------------------------------------- | -------- |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 1        | 0.999    |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 8        | 1.004    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 1        | 0.989    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 8        | 1        |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 1 | 0.999    |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 8 | 1.001    |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *



View result details @ https://www.ytelses.com/opm/?page=result&id=1116

**Comment by totto82** [comment]
@alfbr Any comment on this. Is 1e-12 days a reasonable cutoff?

**Comment by alfbr** [comment]
Yes, I cannot imagine a case where time steps shorter than 1e-12 makes sense.

**Comment by totto82** [comment]
@bska Any other comments on the implementation? It would be great if this can be merged before the weekend. I usually start simulations before the weekend and sometimes they still run on Monday with epsilon timestep size.

**Comment by bska** [comment]
> Any other comments on the implementation?

None.  This looks good to me.  I'll merge into master.

**Pull request merged**